### PR TITLE
invitations: rework for the groups invitations

### PR DESCRIPTION
**Pull request by anikachurilova**
* deprecate COMMUNITIES_GROUPS_ENABLED (moved to USERS_RESOURCES_GROUPS_ENABLED)
* closes https://github.com/inveniosoftware/invenio-app-rdm/issues/2592

Results counter:
Screenshot 2024-04-25 at 11 00 01

Groups tab:
Screenshot 2024-04-25 at 18 08 29
Screenshot 2024-04-24 at 17 13 39

Screenshot 2024-04-24 at 17 10 51
Screenshot 2024-04-24 at 17 11 13
Screenshot 2024-04-24 at 17 11 42

Changes in the **community add members** tab:
Screenshot 2024-04-24 at 17 12 27

Screenshot 2024-04-16 at 15 48 49

### CODE REVIEW

File: invenio_communities/assets/semantic-ui/js/invenio_communities/members/invitations/invitationsModal/GroupTabPane.js [hunk 1]
```diff
@@ -112,12 +112,11 @@ export class GroupTabPane extends Component {
...
             </Form.Field>
-            <Form.Field disabled>
-              <>
-                <label>{i18next.t("Message")}</label>
-                <RichEditor />
-              </>
-            </Form.Field>
+            <i>
+              {i18next.t(
+                "Note: upon addition, selected groups will become community members immediately without any kind of notification."
```
Comment (kpsherva) [inline invenio_communities/assets/semantic-ui/js/invenio_communities/members/invitations/invitationsModal/GroupTabPane.js:117]: minor:
```suggestion
                "Note: upon addition, selected groups will become community members immediately without any kind of notification or invitation approval."
```

Comment (anikachurilova) [reply invenio_communities/assets/semantic-ui/js/invenio_communities/members/invitations/invitationsModal/GroupTabPane.js:117]: Thanks for noticing!
I'm not sure the confirmation modal is necessary here, as we are not doing something dangerous that can not be undone. After closing the modal, user can immediately remove groups access if they did something wrong. What do you think?

File: invenio_communities/config.py [hunk 2]
```diff
@@ -34,6 +34,8 @@
 
 """Communities ui endpoints."""
 
+
+# Deprecated. Please use USERS_RESOURCES_GROUPS_ENABLED config instead.
```
Comment (ntarocco) [inline invenio_communities/config.py:38]: Maybe you are right with simply removing it, as it is not taken into account anymore.
And, we document it well in the changes for v12 (can you please add this in the changes?)

**Review by ntarocco — COMMENTED** [review]
Looks good to me! Thank you for these changes

**Pull request merged**